### ISSUE TITLE
Updating Jenkins

### ISSUE THREAD

**Issue by xtechgamer735** [openshift/jenkins-cloud-plugin#50]
Hey,

Spent the last two hours trying to find out how to update Jenkins on OpenShift and I've given in. Does anyone know how I can do this? SSH wont work and it wont let me update from in app.

Thanks!

**Comment by bparees** [comment]
update what, the version of jenkins itself?

**Comment by xtechgamer735** [comment]
Yeah - the actual version of Jenkins.

**Comment by bparees** [comment]
are you using the openshift image (openshift/jenkins-1-centos7)?  if so, you'd need to build your own image, the version is baked into the image.

you can fork the repo here:
https://github.com/openshift/jenkins
and modify the Dockerfile to install whatever version you want:
https://github.com/openshift/jenkins/blob/master/1/Dockerfile#L31

and then use your image instead of the openshift one.

Note that we install jenkins via rpm and use the LTS releases, you'd need to change the yum install to instead download the latest jenkins tarball and extract it.

Possibly you could do that as a layer on top of the existing image as well.

**Comment by xtechgamer735** [comment]
Yes I was using your image for it. I may have a look at changing it though thank you. One final query, is it possible only connect via HTTP? Since when I use my custom domain as an alias I get certificate errors. 

Thanks

**Comment by bparees** [comment]
that's baked into the config which you can alter when you are also updating the version:
https://github.com/openshift/jenkins/blob/master/1/contrib/openshift/configuration/config.xml.tpl

Note that our jenkins image is also an s2i enabled image, meaning you can also supply your own configuration via the s2i build process as described here:
https://github.com/openshift/jenkins#installing-using-s2i-build

but that won't help you with changing the version.

**Comment by xtechgamer735** [comment]
Thank you for the information - however i am still unsure how I should go about disabling HTTPS. Please could you clarify how I should do this?

Thanks!

**Comment by bparees** [comment]
by default the image is not using https.  how are you deploying and accessing it?

**Comment by xtechgamer735** [comment]
Everything is as it came when I installed it. When I go to my .rhcloud.com sub domain it always forces https and redirects me if I attempt to access it via http. Then when I try to access it through my custom alias the same thing happens and I predictably get SSL certificate mismatch errors.

**Comment by bparees** [comment]
Sorry, I should have realized you were referring to the v2 environment and not v3.  You can ignore my comments about images.

You can supply your own jenkins war file via the jenkins war path env variable:
https://github.com/openshift/origin-server/blob/master/cartridges/openshift-origin-cartridge-jenkins/bin/control#L76

As for the ssl config, i think you need to provide your custom cert to the haproxy instance running on your gear, @dinhxuanvu might be able to help you with that.

**Comment by xtechgamer735** [comment]
Ah thank you for your help Ben! I'll have a look at it later on. I was considering adding my own SSL certificates but I figured trying to disable HTTPS may be easier.

Thanks!
